block/models.py

```diff
@@ -1651,6 +1670,31 @@ def init_menuitem(self, menu, slug, title, order, link, parent=None):
             )
         return menuitem
 
+    def menuitem_list(self, menu_slug):
+        """Return queryset with items with no parent first.
+           Order of items returned is:
+             top level menu items
+             Sub menu items (menus ordered as per their order in top level
+             menu)
+        """
+        menu_items = MenuItem.objects.select_related().filter(
+            menu__slug=menu_slug
+        ).order_by('parent', 'order', 'title')
+        qs = menu_items.extra(
+            select={'parent_null': 'block_menuitem.parent_id is not null'}
```

Comment:
Do we need to use `extra`?  We have managed to avoid all database specific code up to now.  Are we going to start using lots of it?